packages/aws-cdk/lib/settings.ts

```diff
@@ -362,3 +361,20 @@ function expandHomeDir(x: string) {
   }
   return x;
 }
+
+/**
+ * Return all context value that are not an objecg with a _dontSaveContext member
+ */
+function stripTransientValues(obj: any) {
+  const ret: any = {};
+  for (const [key, value] of Object.entries(obj)) {
+    if (!isTransientValue(value)) {
+      ret[key] = value;
+    }
+  }
+  return ret;
+}
+
+function isTransientValue(value: any) {
+  return typeof value === 'object' && value !== null && (value as any)._dontSaveContext;
```

Comment:
`value` might be `undefined` here - and you'll then try to `value._dontSaveContext` which will raise.

Reply:
`typeof undefined` will not be equal to `'object'`.